src/App.tsx

```diff
@@ -1,26 +1,74 @@
+import { useCallback, useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
-import { Loader, TodoFilter, TodoList, TodoModal } from './components';
 
-export const App = () => (
-  <>
-    <div className="section">
-      <div className="container">
-        <div className="box">
-          <h1 className="title">Todos:</h1>
+import { TodoList } from './components/TodoList';
+import { TodoFilter } from './components/TodoFilter';
+import { Todo } from './types/Todo';
+import { TodoModal } from './components/TodoModal';
+import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { filterTodos } from './utils/filterTodos';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { todosSlice } from './features/todos';
 
-          <div className="block">
-            <TodoFilter />
-          </div>
+export const App = () => {
+  const todos = useAppSelector(state => state.todos);
+  const todoDispatch = useAppDispatch();
+  const selectedTodo = useAppSelector(state => state.currentTodo);
+  const { query, category: todoCategory } = useAppSelector(
+    state => state.filter,
+  );
+
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+
+  const setTodos = useCallback(
+    (newTodos: Todo[]) => {
+      todoDispatch(todosSlice.actions.changeTodos(newTodos));
+    },
+    [todoDispatch],
+  );
+
+  const setTodosFromApi = useCallback(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => alert('Todos api is wrong!'))
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }, [setTodos]);
+
+  useEffect(() => {
+    setTodosFromApi();
+  }, [setTodosFromApi]);
+
+  useEffect(() => {
+    setVisibleTodos(() => {
+      return filterTodos(todos, todoCategory, query);
+    });
+  }, [todos, todoCategory, query]);
+
+  return (
+    <>
+      <div className="section">
+        <div className="container">
+          <div className="box">
+            <h1 className="title">Todos:</h1>
+
+            <div className="block">
+              <TodoFilter />
+            </div>
 
-          <div className="block">
-            <Loader />
-            <TodoList />
+            <div className="block">
+              {isLoading && <Loader />}
+              {!isLoading && <TodoList todos={visibleTodos} />}
+            </div>
           </div>
         </div>
       </div>
-    </div>
 
-    <TodoModal />
-  </>
-);
+      {selectedTodo && <TodoModal />}
```

Comment:
Ensure that the `TodoModal` component is correctly handling the `selectedTodo` prop or state. If `selectedTodo` is required by `TodoModal`, make sure it is being passed correctly.